lib/pharos/phases/validate_host.rb

```diff
@@ -43,10 +43,13 @@ def gather_host_facts
       end
 
       def check_role
-        return if @host.master? && @host.checks['ca_exists'] == true
-        return if @host.worker? && @host.checks['kubelet_configured'] && @host.checks['ca_exists'] != true
+        return if !@host.checks['kubelet_configured']
```

Comment:
This will not work correctly if the cluster setup failed after the `ConfigureKubelet`/`ConfigureEtcd` phases but before the `ConfigureMaster`/`JoinNode` phases, and the role was then changed... Might be better to check for the `/etc/kubernetes/manifests/*` files instead ( `pharos-proxy`, `pharos-etcd` and `kube-apiserver`), since AFAIK those are what actually cause the actual role incompatibilities per #305.